### PR TITLE
Make src/script comply with the style guide

### PR DESCRIPTION
**Pull request by slashrsm**

### CODE REVIEW

File: src/script/script.js [hunk 1]
```diff
@@ -105,249 +104,240 @@ Script.fromBuffer = function(buffer) {
...
+    let opcodenum = opcode.toNumber();
 
     if (_.isUndefined(opcodenum)) {
-      opcodenum = parseInt(token);
+      // #wierdstuff
+      if (token.startsWith('0x')) {
+        opcodenum = parseInt(token, 16);
+      } else {
+        opcodenum = parseInt(token, 10);
+      }
```
Comment (slashrsm) [inline src/script/script.js]: Yup, wierd.

File: src/script/script.js [hunk 2]
```diff
@@ -458,12 +445,16 @@ Script.prototype.isDataOut = function() {
...
  */
-Script.prototype.getData = function() {
+Script.prototype.getData = function () {
   if (this.isDataOut() || this.isScriptHashOut()) {
-    return new Buffer(this.chunks[1] === undefined ? 0 : this.chunks[1].buf);
+    // #wierdstuff
+    if (this.chunks[1] === undefined) {
+      return Buffer.alloc(0);
+    }
+    return Buffer.from(this.chunks[1].buf);
```
Comment (slashrsm) [inline src/script/script.js]: I've seen this a few times already. Weird.

File: src/script/script.js [hunk 3]
```diff
@@ -708,19 +697,22 @@ Script.buildMultisigOut = function(publicKeys, threshold, opts) {
...
+ * @param {boolean=} opts.noSorting don't sort the given public keys before creating the script
+ *   (false by default)
  * @param {Script=} opts.cachedMultisig don't recalculate the redeemScript
  *
  * @returns {Script}
  */
-Script.buildMultisigIn = function(pubkeys, threshold, signatures, opts) {
+// #wierdstuff - unused function arg
+// eslint-disable-next-line no-unused-vars
+Script.buildMultisigIn = function (pubkeys, threshold, signatures, opts) {
```
Comment (slashrsm) [inline src/script/script.js]: But tests go red if I remove it...

File: src/script/interpreter.js [hunk 4]
```diff
@@ -893,40 +864,40 @@ Interpreter.prototype.step = function() {
...
+            break;
+          case Opcode.OP_NOT:
+            bn = new BN((bn.cmp(BN.Zero) === 0) + 0);
+            break;
+          case Opcode.OP_0NOTEQUAL:
+            bn = new BN((bn.cmp(BN.Zero) !== 0) + 0);
+            break;
+          default:
+            // Throw an error?
+            break;
```
Comment (slashrsm) [inline src/script/interpreter.js]: default clause is required by eslint. What to do here?

File: src/script/interpreter.js [hunk 5]
```diff
@@ -942,81 +913,81 @@ Interpreter.prototype.step = function() {
...
+            bn = new BN((bn1.cmp(bn2) >= 0) + 0);
+            break;
+          case Opcode.OP_MIN:
+            bn = (bn1.cmp(bn2) < 0 ? bn1 : bn2);
+            break;
+          case Opcode.OP_MAX:
+            bn = (bn1.cmp(bn2) > 0 ? bn1 : bn2);
+            break;
+          default:
+            // Throw an error?
```
Comment (slashrsm) [inline src/script/interpreter.js]: Same as above.

File: src/script/script.js (hunk 1 above)
Comment (ClemensLey) [reply src/script/script.js]: Great to catch this

**Comment by ClemensLey** [comment]
This is great work Janez!

**Pull request merged**